pkg/prometheus/agent/operator.go

```diff
@@ -292,12 +292,12 @@ func (c *Operator) Run(ctx context.Context) error {
 		return nil
 	}
 
-	allowed, err := c.checkAgentAuthorization(ctx)
+	missingPermissions, err := c.getMissingPermissions(ctx)
 	if err != nil {
 		return err
 	}
-	if !allowed {
-		level.Info(c.logger).Log("msg", "Prometheus agent controller disabled because it lacks the required permissions on PrometheusAgent objects")
+	if missingPermissions != nil {
+		level.Info(c.logger).Log("msg", "Prometheus agent controller disabled because it lacks the required permissions on PrometheusAgent objects", "missingpermissions", fmt.Sprintf("%v", missingPermissions))
```

Comment:
Should this be a Warn? I kinda feel that this is more important than just info 🤔 but I'm also fine with `Info`

Reply:
Good point!